src/core_ghidra.cpp

```diff
@@ -132,109 +133,120 @@ static void ApplyPrintCConfig(RConfig *cfg, PrintC *print_c)
-		}
-
-		switch (mode)
-		{
-			case DecompileMode::XML:
-			case DecompileMode::DEFAULT:
-			case DecompileMode::JSON:
-			case DecompileMode::OFFSET:
-			case DecompileMode::STATEMENTS:
-				arch.print->setXML(true);
-				break;
-			default:
-				break;
-		}
-
-		if(mode == DecompileMode::XML)
+		action->reset(*func);
+		res = action->perform(*func);
+	}
+	catch(const LowlevelError &error)
+	{
+		arch.getCore()->sleepEndForce();
+		throw error;
+	}
+	arch.getCore()->sleepEnd();
+	if (res<0)
+	{
+		eprintf("break\n");
+	}
+	if(cfg_var_verbose.GetBool(core->config))
+	{
+		for(const auto &warning : arch.getWarnings())
 		{
-			out_stream << "<result><function>";
-			func->saveXml(out_stream, 0, true);
-			out_stream << "</function><code>";
+			func->warningHeader("[r2ghidra] " + warning);
 		}
-
-		RAnnotatedCode *code = nullptr;
-		switch(mode)
-		{
-			case DecompileMode::XML:
-			case DecompileMode::DEFAULT:
-			case DecompileMode::JSON:
-			case DecompileMode::OFFSET:
-			case DecompileMode::STATEMENTS:
-				arch.print->docFunction(func);
-				if(mode != DecompileMode::XML)
+	}
+	switch (mode)
+	{
+		case DecompileMode::XML:
+		case DecompileMode::DEFAULT:
+		case DecompileMode::JSON:
+		case DecompileMode::OFFSET:
+		case DecompileMode::STATEMENTS:
+			arch.print->setXML(true);
+			break;
+		default:
+			break;
+	}
+	if(mode == DecompileMode::XML)
+	{
+		out_stream << "<result><function>";
+		func->saveXml(out_stream, 0, true);
+		out_stream << "</function><code>";
+	}
+	switch(mode)
+	{
+		case DecompileMode::XML:
+		case DecompileMode::DEFAULT:
+		case DecompileMode::JSON:
+		case DecompileMode::OFFSET:
+		case DecompileMode::STATEMENTS:
+			arch.print->docFunction(func);
+			if(mode != DecompileMode::XML)
+			{
+				code = ParseCodeXML(func, out_stream.str().c_str());
+				if (!code)
 				{
-					code = ParseCodeXML(func, out_stream.str().c_str());
-					if (!code)
-						throw LowlevelError("Failed to parse XML code from Decompiler");
+					throw LowlevelError("Failed to parse XML code from Decompiler");
 				}
-				break;
-			case DecompileMode::DEBUG_XML:
-				arch.saveXml(out_stream);
-				break;
-			default:
-				break;
-		}
+			}
+			break;
+		case DecompileMode::DEBUG_XML:
+			arch.saveXml(out_stream);
+			break;
+		default:
+			break;
+	}
+}
 
+RAnnotatedCode* r2ghidra_decompile_annotated_code(RCore *core, ut64 addr){
+	DecompilerLock lock;
+	RAnnotatedCode *code = nullptr;
+	try
+	{
+		std::stringstream out_stream;
+		refactored_decompile(core, out_stream, DecompileMode::DEFAULT, code, addr);
+		return code;
+	}
+	catch(const LowlevelError &error)
+	{
+		std::string s = "Ghidra Decompiler Error: " + error.explain;
+		char *err = strdup (s.c_str());
+ 		code = r_annotated_code_new(err);
+		// Push an annotation with: range = full string, type = error
+		// For this, we have to modify RAnnotatedCode to have one more type; for errors
+		return code;
+	}
+}
+
+static void Decompile(RCore *core, DecompileMode mode)
```

Comment:
```suggestion
static void DecompileCmd(RCore *core, DecompileMode mode)
```
to distinguish it from the regular `Decompile()` as mentioned above.

Reply:
Done.